### DISCUSSION TITLE
Way to mass-edit attachments from private to public?

### DISCUSSION THREAD

**Discussion by mindsolve** [Part-DB/Part-DB-server#292] [Q&A]
I've imported all data from a previous PartKeepr installation, but I've noticed, that all attachments are marked as "private" and was not able to find some kind of mass-edit function for attachments nor a config option during import for that.

Is there a way to mass-mark attachments as not private?

**Comment by jbtronics** [comment]
In general its intended behavior, that most files should be private, so that you can control access to them, and only users which are logged in can access them (The public files can be accessed by anybody who knows the URL or you can even try to guess them).

The only exception where "public" files are useful are pictures, as we then can create thumbnails for them and can use the more performant delivery directly via the webserver without invoking PHP at all. In principle you could also create thumbnails for "private" files, but as the thumbnails are always reachable via the public URL, that would be a bit pointless.

**Reply by mindsolve** [reply]
Just to be sure I understand you correctly: The disabled option of viewing/downloading attachments (when logged in) is not something related to the fact that the attachement is private? Shall I open an issue for that?
![grafik](https://github.com/Part-DB/Part-DB-server/assets/24863329/df1f6720-227e-4e44-9e72-ba6238bea6bf)


Also - for me at least - there are no attachments that are somehow "secret". They are 100% (public) datasheets and component photos. So I would absolutely want to give anonymous access to them. But for a future, where I might actually have attachments that are supposed to be only visible to logged in users, giving everybody the "access private attachments" isn't something I would want to do.
So a "one time" way for mass-changing this status would be very appreciated.

**Reply by jbtronics** [reply]
That the buttons for download and view are greyed seems to be a bug. The access to the attachment via the controller is possible like intended, only the buttons seems to be rendered incorrectly.
